src/rules/unique-dependencies.ts

```diff
@@ -19,9 +19,14 @@ const dependencyPropertyNames = new Set([
 	"peerDependencies",
 ]);
 
+const requiredUniqueDependencies = new Set(["dependencies", "devDependencies"]);
```

Comment:
Similar to the other comment, this name doesn't really do much to explain its purpose.  Maybe something like `groupsRequiringGlobalUniqueness`?